### ISSUE TITLE
option to enforce messages on `assert` statements

### ISSUE THREAD

**Issue by DetachHead** [PyCQA/pylint#5328]
### Current problem

it's easy to write an `assert` statement that results in a useless error message:

```py
assert expected_value == actual_value
```
if this assertion fails, it will give you a useless error message like `AssertionError` with no other information.

### Desired solution

a pylint rule that enforces a message argument to the assert statement, ie:
```py
assert expected_value == actual_value, f"{expected_value=} did not match {actual_value=}"
```

**Comment by Pierre-Sassoulas** [comment]
It would needs to be an optional extension because this won't work well with code base using pytest (that already add a helpful error message in most case).

**Comment by cdce8p** [comment]
Just an idea. Maybe we can check either the method or module name before emitting a warning. With the exception of delegated test handlers, they usually start with `test_*`. It should be easy enough to ignore warnings in these cases.

**Comment by cdce8p** [comment]
One thing to be aware of: `assert` can also be used for type narrowing. We just added that in an `astroid` PR: https://github.com/PyCQA/astroid/pull/1244#discussion_r751174409. If possible, would be nice if that could be detected as well.

```py
class X:
    def __init__(self, var: str | None) -> None:
        self.var = var

def func(a: X) -> None:
    assert a.var
    print(a.var.lower())

func(X("Hello World"))
```
